lib/ansible/modules/cloud/vmware/vmware_host_dns.py

```diff
@@ -0,0 +1,234 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2018, Christian Kotte <christian.kotte@gmx.de>
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = r'''
+---
+module: vmware_host_dns
+short_description: Manage DNS configuration of an ESXI host system
+description:
+- This module can be used to configure DNS for ESXi host systems in given vCenter infrastructure.
+- DNS can only be configured for the default TCP/IP stack.
+version_added: 2.8
+author:
+- Christian Kotte (@ckotte) <christian.kotte@gmx.de>
+notes:
+- Tested on vSphere 6.5
+requirements:
+- python >= 2.6
+- PyVmomi
+options:
+  host_name:
+      description:
+         - Host name
+      required: True
+  domain:
+      description:
+         - Domain
+      required: True
+  dns_servers:
+      description:
+         - DNS servers
+      required: True
+  search_domains:
+     description:
+        - Search domains
+     required: True
+  esxi_hostname:
+    description:
+    - Name of the host system to work with.
+    - This is required parameter if C(cluster_name) is not specified.
+  cluster_name:
+    description:
+    - Name of the cluster from which all host systems will be used.
+    - This is required parameter if C(esxi_hostname) is not specified.
+extends_documentation_fragment: vmware.documentation
+'''
+
+EXAMPLES = r'''
+- name: Configure DNS for an host system
+  vmware_host_dns:
+    hostname: '{{ vcenter_hostname }}'
```

Comment:
We prefer that examples show sensible values, so people understand what is expected.
If you add variables like this, there's no clue what the actual content would be.

Reply:
Ok, I fixed it.

Reply:
I would keep real values there, and honestly it would be best to standardize this over the various modules as well. Something we need to agree to in the VMware WG. See: https://github.com/ansible/ansible/issues/43433
```yaml
  hostname: vcenter01.fqdn.local  # Would be nice to have a standard example domain
  username: Administrator@vsphere.local  # This could be the default username, so it's optional
  password: SuperSecretPassword  # Also a standard password would be good
  esxi_hostname: esx01.fqdn.local
  host_name: vmguest01  # I guess in some examples we could use a variable though
```

Reply:
The aim of having a `vcenter` connection plugin, would be that the connection information of the guest would store the vcenter info/credentials, so that we wouldn't need to repeat this information. And we should probably use the same connection variables for both the `vcenter` and `vmware_tools` connection plugins. cc @dericcrago

Reply:
> The aim of having a `vcenter` connection plugin, would be that the connection information of the guest would store the vcenter info/credentials, so that we wouldn't need to repeat this information. And we should probably use the same connection variables for both the `vcenter` and `vmware_tools` connection plugins. cc @dericcrago

How would this work? Do I need to run the connection module at the beginning of the playbook and then at the end again to disconnect? Similar to `Connect-VIServer` and `Disconnect-VIServer` cmdlets in PowerShell scripts?

Reply:
> I would keep real values there, and honestly it would be best to standardize this over the various modules as well. Something we need to agree to in the VMware WG. 

Yes, the documentation is not consistent as well as the name of certain options. Thanks for pointing that out. Unfortunately, we didn't discuss standardization during our last WG.